### PR TITLE
convert client-side tests to Splinter

### LINKED ISSUE #3684: Use Splinter for web tests
Currently I don't know who knows how our web test work, because if is based on JavaScript stack. The Splinter allows to write the same stuff in Pythonic way.

https://splinter.readthedocs.org/en/latest/index.html

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/25853286-use-splinter-for-web-tests?utm_campaign=plugin&utm_content=tracker%2F85909&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F85909&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by whit537**
Closes #3684.

I wanted to write client-side tests for #4077, but I wanted to write more than just a few of them, and I didn't want to write them in our current client-side testing framework.

### CODE REVIEW

**Comment by whit537** [comment]
Yessssss. ba872d4d9cb6848fbb6ffb5983f9c44104f3fa2e passes Travis! 👍 💃

**Comment by whit537** [comment]
Ready for review, @aandis @rohitpaulk! :-)

File: tests/test.env [hunk 1]
```diff
@@ -1,5 +1,6 @@
 DATABASE_URL="dbname=gratipay-test"
-BASE_URL=
+BASE_URL=""
```
Comment (aandis) [inline tests/test.env]: Let's remove this?

Comment (whit537) [inline tests/test.env]: Done in 1f27eaf8f613e2ffb057e1252cea8a50f339616d.

**Comment by aandis** [comment]
Rebased on master. Previous head was 1f27eaf8f613e2ffb057e1252cea8a50f339616d

**Comment by aandis** [comment]
Looks good to me @whit537 . ready for merge..

**Comment by aandis** [comment]
Let's wait for [github](https://status.github.com/messages/2016-07-12) to get back to normal?

**Comment by whit537** [comment]
Cool. Looks like GitHub is [green](https://status.github.com/messages/2016-07-12) again!

**Comment by whit537** [comment]
Reopening to bump Travis.

**Comment by whit537** [comment]
Alright! We're green. :-)

**Pull request merged**